src/Workspaces/Core/Portable/Shared/TestHooks/TaskExtensions.cs

```diff
@@ -45,5 +46,122 @@ static Task CompletesTrackingOperationSlow(Task task, IDisposable token)
                     TaskScheduler.Default);
             }
         }
+
+        // Following code is copied from Microsoft.VisualStudio.Threading.TplExtensions
+        // https://github.com/microsoft/vs-threading/blob/main/src/Microsoft.VisualStudio.Threading/TplExtensions.cs
+
+        /// <summary>
+        /// Returns an awaitable for the specified task that will never throw, even if the source task
+        /// faults or is canceled.
+        /// </summary>
+        /// <param name="task">The task whose completion should signal the completion of the returned awaitable.</param>
+        /// <param name="captureContext">if set to <c>true</c> the continuation will be scheduled on the caller's context; <c>false</c> to always execute the continuation on the threadpool.</param>
+        /// <returns>An awaitable.</returns>
+        public static NoThrowTaskAwaitable NoThrowAwaitable(this Task task, bool captureContext = true)
+        {
+            return new NoThrowTaskAwaitable(task, captureContext);
+        }
+
+        /// <summary>
+        /// An awaitable that wraps a task and never throws an exception when waited on.
+        /// </summary>
+        public readonly struct NoThrowTaskAwaitable
```

Comment:
own file?

Reply:
I think it might be better to keep copied code together? But I'm fine either way.

Reply:
if you prefer that, then that's fine :)